### PR TITLE
Ports: Enable running python3 when built with clang

### PR DESCRIPTION
**Pull request by ADKaster**
Proof is in the pudding:
python-clang

Patches to:

- sqlite: Let it build shlibs without the old copy whole archive hack
- openssl: add /usr/local/lib with -L
- top lvl: Add READELF define to un-confuse the python build script when there's no x86_64-pc-serenity-readelf.
- python: Don't try to enable PGO
- LibCrypt: Remove circular library dependency

### CODE REVIEW

File: Ports/python3/package.sh [hunk 1]
```diff
@@ -17,14 +17,16 @@ icon_file="../launcher.ico" # This is an older icon that's downloaded separately
...
-configopts=("--enable-optimizations" "--disable-ipv6" "--without-ensurepip" "ac_cv_file__dev_ptmx=no" "ac_cv_file__dev_ptc=no")
+configopts=("--disable-ipv6" "--without-ensurepip" "ac_cv_file__dev_ptmx=no" "ac_cv_file__dev_ptc=no")
 
-export CC="${CC} --sysroot=${SERENITY_INSTALL_ROOT}"
 export BLDSHARED="${CC} -shared"
 
 pre_configure() {
     build="$("${workdir}/config.guess")"  # e.g. 'x86_64-pc-linux-gnu'
     configopts+=("--build=${build}")
+    if [ "$SERENITY_TOOLCHAIN" = "GCC" ]; then
```
Comment (linusg) [inline Ports/python3/package.sh:27]: Please add a comment

**Review by linusg — COMMENTED** [review]
Typo in second commit: pacakges

File: Ports/python3/package.sh [hunk 2]
```diff
@@ -17,14 +17,16 @@ icon_file="../launcher.ico" # This is an older icon that's downloaded separately
...
+        configopts+=("--enable-optimizations")
+    fi
```
Comment (BertalanD) [inline Ports/python3/package.sh:29]: Does `--enable-optimizations` have any effect when building with GCC? If not, leaving this here would be misleading to the reader. 

If I'm reading Python's `configure` correctly,  `--enable-optimizations` controls PGO and `-fno-semantic-interposition`. We are definitely not taking advantage of PGO with either of the two toolchains, and our GCC toolchain has `-fno-semantic-interposition` enabled by default.

Comment (ADKaster) [reply Ports/python3/package.sh:29]: Ah yes, looking at the README.rst for 3.10.1 , --enable-optimizations currently simply enables PGO. LTO is enabled by a different flag (we could maybe try that one :thonk:).

We won't be able to take advantage of this flag unless we build on-target. I'll remove it completely instead.

File: Ports/python3/package.sh (hunk 1 above)
Comment (ADKaster) [reply Ports/python3/package.sh]: Typo in review text addressed, comment itself is OBE, ditching this flag entirely instead :)

**Pull request merged**